app/pages/resource/ResourcePageContainer.js

```diff
@@ -1,37 +1,60 @@
+import queryString from 'query-string';
 import React, { Component, PropTypes } from 'react';
 import Loader from 'react-loader';
 import { connect } from 'react-redux';
+import { browserHistory } from 'react-router';
 
 import { fetchResource } from 'api/actions';
 import ResourcePage from './ResourcePage';
 import selector from './resourcePageSelector';
 
 export class UnconnectedResourcePageContainer extends Component {
   static propTypes = {
+    date: PropTypes.string.isRequired,
     fetchResource: PropTypes.func.isRequired,
     isLoaded: PropTypes.bool.isRequired,
+    location: PropTypes.shape({
+      query: PropTypes.object.isRequired,
+    }).isRequired,
     params: PropTypes.shape({
       id: PropTypes.string.isRequired,
     }).isRequired,
     resource: PropTypes.object,
     unit: PropTypes.object,
   };
 
+  constructor(props) {
+    super(props);
+    this.handleDateChange = this.handleDateChange.bind(this);
+  }
+
   componentDidMount() {
     this.props.fetchResource(this.props.params.id);
   }
 
+  handleDateChange(date) {
+    const query = queryString.stringify({
+      ...this.props.location.query,
```

Comment:
I'm wondering what would be a good general way to handle cases where we want to only update one query paremeter in the url. Accessing `this.props.location.query` only works on Route level Components and adding the thing to propTypes is somewhat of a hassle. 
Something like `...browserHistory.getCurrentLocation().query` would work, but not sure if it has some downsides or is an un-reactish way to do things.